From d7ce032a906775a850f3b330e29101be004dcf6b Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 10 May 2022 10:03:47 +0200
Subject: [PATCH 0041/1695] core: kernel: tee_ta_manager.c: add uuid in open
 session error trace

Adds the TA UUID in open session error trace to allow to identify
witch TA cause the issue when debug trace are not acitvated.

By the way, fix specifier for res argument that is a uint32_t.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fd3f27c651a8f4c244406007d5abdb9039101411 core: kernel: tee_ta_manager.c: add uuid in open session error trace]
Change-Id: I18ae7c79eb06e4d4f737f577fdea4f65d6a2d400
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
[etienne: cherry-picked from 3d8cac149c38dfce2cfeb87b62ce917094affb2f]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/kernel/tee_ta_manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/kernel/tee_ta_manager.c b/core/kernel/tee_ta_manager.c
index 97dee430c..645eaefe6 100644
--- a/core/kernel/tee_ta_manager.c
+++ b/core/kernel/tee_ta_manager.c
@@ -740,7 +740,7 @@ TEE_Result tee_ta_open_session(TEE_ErrorOrigin *err,
 	if (!res)
 		*sess = s;
 	else
-		EMSG("Failed. Return error 0x%x", res);
+		EMSG("Failed for TA %pUl. Return error %#"PRIx32, uuid, res);
 
 	return res;
 }
-- 
2.47.2

